datasets/flwr_datasets/partitioner/non_iid_partitioner.py

```diff
@@ -0,0 +1,68 @@
+from typing import Dict
+import datasets
+from flwr_datasets.partitioner.natural_id_partitioner import NaturalIdPartitioner
+import numpy as np
+
+class NonIidPartitioner(NaturalIdPartitioner):
+    """Partitioner for a dataset in which the final partition results in different 
+        non overlapping datasets for a given number of groups/nodes"""
+
+    def __init__(self, partition_by, num_nodes) -> None:
```

Comment:
Happy to change that, to me what makes the most sense is the following. (For simplicity let's say that each class of the dataset has 1000 elements, and there are 10 classes).

If `group_size`= 1000 -> 10 groups are created one for each label (Same behavior as NaturalIDPartitioner)

If `group_size` < 1000 -> 
   For each label we have n= 1000 / {{group_size}} groups. 
   Case 1: There is no reminder: Divide the given data with a IID Partitioner among the n groups.
   Case 2: There is a reminder: I would may change `mode` to a boolean `drop` by default false. If true, we drop the reminder and  not use it. If false, we add it to the last group.

If `group_size` > 1000 ->
  We have 1000 * 10 / `group_size` groups:
  Case 1: There is no reminder. Use the current code.
  Case 2: There is reminder, I would do the same as in numer 1.  Unsure if rounding up, or down to decide the number 
 of groups.

Reply:
Do you agree with this way of doing it? If not I am able to do it in another way.

Reply:
@CarlosMari Let's split this PR into two. This one with `GroupedNaturalIdPartitioner` and the other with `NumNodesGroupedNaturalIdPartitioner`. However, I'm not that convinced about the usability of the second one. We can have the discussion in the next PR.
As for `GroupedNaturalIdPartitioner`, it'd be nice it the creation of the object would be possible when specyfing `group_size` also (as an alternative to `num_groups`). I see two ways of doing that 1) add both parameters to the init and make them optional (with internal logic of being mutually exclusive) 2) make an alternative constructor using `@classmethod `. @jafermarq do you think would be a cleaner solution?

Reply:
Ok! Will get to that soon!

Reply:
@adam-narozniak Hi! I have added the group_size part you requested. As it is supposed to be on the non-IID side, it sorts the dataset by label and depending on group_size it gets the results. 

For example: 
`group_size` = 20 , `data_per_label` = 31, `num_labels` = 2
The first node would get the first label [0-19], second one would get [20,39] (items of both labels) and last one [39,61] 
(If the remaining data is less than 50% of group_size it gets added to last group, otherwise a smaller group is created)

It is completely up to you to decide, but the original idea I had with this partitioner was to make it completely non-iid, so I feel like this size-non-iid and group nodes partitioner do not really have the same behavior and should be different classes. I have not yet gotten to tests and formatting as I wanted to check it this is what you had in mind.

Reply:
I would probably see two different partitioners for non iid data, this one where you input group_size, and the one I had previously committed (currently not in a PR) where you select a number of non_iid groups and a total number of nodes.

Reply:
Hi, the last changes got me quite lost. I'm gonna follow up on the work from your initial commit.